[new-product] KEDA

nunoadrego opened the pull request:
The EOL policy is not detailed. I [reached out](https://kubernetes.slack.com/archives/CKZJ36A5D/p1690841228601669) in the `#keda` Kubernetes Slack channel.

nunoadrego committed:
[new-product] KEDA

nunoadrego committed:
Fix changelogTemplate

captn3m0 commented on products/keda.md:
```diff
@@ -0,0 +1,70 @@
+
+identifiers:
+  - purl: pkg:github/kedacore/keda
+  - purl: pkg:github/kedacore/charts
```
1. Lets add the OCI images as well. (Based this on the [official release manifest](https://github.com/kedacore/keda/releases/download/v2.11.2/keda-2.11.2.yaml))
2. https://github.com/kedacore/charts/releases shows a few releases where chart versioning does not match the keda version, so not sure if we should be using that.

```suggestion
  - purl: pkg:oci/keda-admission-webhooks?repository_url=ghcr.io/kedacore
  - purl: pkg:oci/keda-metrics-apiserver?repository_url=ghcr.io/kedacore
  - purl: pkg:oci/keda?repository_url=ghcr.io/kedacore
```

nunoadrego replied:
Agree! Thanks.

nunoadrego committed:
Update products/keda.md

Co-authored-by: Nemo <me@captnemo.in>

marcwrobel commented on products/keda.md:
```diff
@@ -0,0 +1,72 @@
+---
+title: KEDA
+category: server-app
+tags: kubernetes
```
```suggestion
tags: cncf
```


`kubernetes` is not an existing tag, not sure we want to add it. @captn3m0 do we want to add this tag ? What are the other products that could use it in such case ?

@nunoadrego just to explain, we want to have a very limited set of meaningful tags, so new one must be added carefully.

Edit: updated suggestion based on below comments

nunoadrego replied:
Other projects that can use this tag:
- Kubernetes
- Azure Kubernetes Service
- Amazon EKS
- Google Kubernetes Engine
- etcd (?)
- Istio

But as @captn3m0 said, a CNCF tag might be another possibility.

captn3m0 replied:
For k8s, we could perhaps use a separate `kubernetes-distribution` tag. (That would just apply to k8s, aks, eks, gke pages). 

cncf here should be good enough.

marcwrobel commented on products/keda.md:
```diff
@@ -0,0 +1,72 @@
+permalink: /keda
+releasePolicyLink: https://keda.sh/docs/latest/operate/cluster/
+changelogTemplate: "https://github.com/kedacore/keda/releases/tag/v__LATEST__"
+eolColumn: false
```
No EOL information ? Would it be possible to give a rough idea based on the latest releases. 

As an example you can have a look at Kafka (https://endoflife.date/apache-kafka), which does not have a documented policy too.

And from what I can see it looks like only the latest release is supported: there seem to be no update once a new release is out.

marcwrobel replied:
I updated the description and added a few more releases. Let me know if it's OK for you.

marcwrobel commented on products/keda.md:
```diff
@@ -0,0 +1,72 @@
+changelogTemplate: "https://github.com/kedacore/keda/releases/tag/v__LATEST__"
+eolColumn: false
+releaseColumn: Latest
+releaseDateColumn: Released
```
```suggestion
releaseDateColumn: true
```

Those labels are the default ones. And `releaseColumn` is displayed by default.

marcwrobel commented on products/keda.md:
```diff
@@ -0,0 +1,72 @@
+
+KEDA follows [Semantic Versioning](https://github.com/kedacore/governance/blob/main/RELEASES.md).
+A new minor version is released approximately once a quarter on Thursdays.
+
```
```suggestion

KEDA does not have a clearly defined support policy, but looking at the recent history it seems only the latest release is supported.
```

captn3m0 committed:
Update products/keda.md

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

captn3m0 committed:
Apply suggestions from code review

Co-authored-by: Marc Wrobel <marc.wrobel@gmail.com>

captn3m0 committed:
[keda] Mark 2.7, 2.8 as EOL

This PR: https://github.com/kedacore/keda/issues/4132
backported security fixes to 2.8.2 as "exceptional". Hence:

- Marked 2.8 as EOL on the date of this exceptional release.
- Marked 2.7 as EOL, on the date of 2.8 release, which seems to have been the case.

From looking at release and backport history, I am guessing 2.9 isn't getting any security backports either, but we can wait for clarity on those.

marcwrobel commented on products/keda.md:
```diff
@@ -0,0 +1,72 @@
+A new minor version is released approximately once a quarter on Thursdays.
+
+KEDA does not have a clearly defined support policy, but looking at the recent history it seems only the latest release is supported.
+## [Supported Kubernetes Versions](https://keda.sh/docs/latest/operate/cluster/#kubernetes-compatibility)
```
```suggestion
KEDA does not have a clearly defined support policy, but looking at the recent history it seems only the latest release is supported.

## [Supported Kubernetes Versions](https://keda.sh/docs/latest/operate/cluster/#kubernetes-compatibility)
```

marcwrobel commented on products/keda.md:
```diff
@@ -0,0 +1,72 @@
+category: server-app
+tags: cncf
+permalink: /keda
+releasePolicyLink: https://keda.sh/docs/latest/operate/cluster/
```
```suggestion
releasePolicyLink: https://github.com/kedacore/keda/blob/main/SECURITY.md
```

marcwrobel replied:
IMO the most important thing for a release / support policy is to be publicly documented. So I think even if it does not match the reality I think we should go ahead with https://github.com/kedacore/keda/blob/main/SECURITY.md:

> KEDA commits to supporting the n-2 version minor version of the current major release; as well as the last minor version of the previous major release.

WDYT ?

marcwrobel committed:
Update products/keda.md

marcwrobel committed:
Update support policy

marcwrobel committed:
Add more versions, improve description

marcwrobel commented on products/keda.md:
```diff
@@ -0,0 +1,133 @@
+
+According to their documentation KEDA [commits](https://github.com/kedacore/keda?tab=security-ov-file#readme)
+to supporting the last two minor versions of the current major release, as well as the last minor version of
+the previous major release.
```
The security policy was updated in https://github.com/kedacore/keda/pull/5049. They removed the _last two minor versions + last minor version of the previous major_ rule and linked to https://keda.sh/support/, which states:

> KEDA is an open-source project under CNCF Foundation and provides best-effort support and use GitHub for tracking bugs and feature requests.

The description must be updated accordingly.

marcwrobel replied:
Added 2.12. Looking at the recent release history it looks like they only support the latest version, except for 2.8 which had an update after the 2.9.0 release.

We could change the releasePolicyLink to https://keda.sh/support/ and the description to:

> KEDA follows [Semantic Versioning](https://github.com/kedacore/governance/blob/main/RELEASES.md). A new minor version is released approximately every three months and is usually supported until the next minor version is released.

WDYT ?

marcwrobel replied:
Applied the update.

marcwrobel committed:
Add 2.12

marcwrobel commented on products/keda.md:
```diff
@@ -0,0 +1,140 @@
+
+According to their documentation KEDA [commits](https://github.com/kedacore/keda?tab=security-ov-file#readme)
+to supporting the last two minor versions of the current major release, as well as the last minor version of
+the previous major release.
```
```suggestion
KEDA follows [Semantic Versioning](https://github.com/kedacore/governance/blob/main/RELEASES.md).
A new minor version is released approximately every three months.

KEDA follows [Semantic Versioning](https://github.com/kedacore/governance/blob/main/RELEASES.md).
A new minor version is released approximately every three months and is usually supported until the
 next minor version is released.
```

marcwrobel committed:
Update products/keda.md

marcwrobel commented on products/keda.md:
```diff
@@ -0,0 +1,140 @@
+
+KEDA follows [Semantic Versioning](https://github.com/kedacore/governance/blob/main/RELEASES.md).
+A new minor version is released approximately every three months.
+
```
```suggestion
```

marcwrobel committed:
Update products/keda.md

marcwrobel commented on products/keda.md:
```diff
@@ -0,0 +1,137 @@
+
+KEDA follows [Semantic Versioning](https://github.com/kedacore/governance/blob/main/RELEASES.md).
+A new minor version is released approximately every three months and is usually supported until the
+ next minor version is released.
```
```suggestion
KEDA follows [Semantic Versioning](https://github.com/kedacore/governance/blob/main/RELEASES.md).
A new minor version is released approximately every three months and is usually supported until the
next minor version is released.
```

marcwrobel committed:
Update products/keda.md

marcwrobel reviewed (approved):
Quickly re-checked. LGTM, merging. If something needs to be changed it will be addressed in a follow-up PR.
